src/sage/rings/semirings/tropical_variety.py

```diff
@@ -0,0 +1,1205 @@
+          ``random``); ``random`` means each polygon will be assigned
+          a different color. If instead a specific ``color`` is provided,
+          then all polygon will be given the same color.
+
+        OUTPUT: Graphics3d Object
+
+        EXAMPLES:
+
+        A tropical surface that consist of only one cell::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x,y,z> = PolynomialRing(T)
+            sage: p1 = x + z
+            sage: tv = p1.tropical_variety()
+            sage: tv.plot()
+            Graphics3d Object
+
+        .. PLOT::
+            :width: 300 px
+
+            T = TropicalSemiring(QQ)
+            R = PolynomialRing(T, ('x,y,z'))
+            x, y, z = R.gen(), R.gen(1), R.gen(2)
+            p1 = x + z
+            sphinx_plot(p1.tropical_variety().plot())
+
+        A tropical surface with multiple cell that exhibit complex and
+        intriguing geometric structures::
+
+            sage: p2 = x^2 + x + y + z + R(1)
+            sage: tv = p2.tropical_variety()
+            sage: tv.plot()
+            Graphics3d Object
+
+        .. PLOT::
+            :width: 300 px
+
+            T = TropicalSemiring(QQ)
+            R = PolynomialRing(T, ('x,y,z'))
+            x, y, z = R.gen(), R.gen(1), R.gen(2)
+            p2 = x**2 + x + y + z + R(1)
+            sphinx_plot(p2.tropical_variety().plot())
+        """
+        from random import random
+        from sage.plot.graphics import Graphics
+        from sage.geometry.polyhedron.constructor import Polyhedron
+
+        if color == 'random':
+            colors = []
+            for _ in range(self.number_of_components()):
+                color = (random(), random(), random())
+                colors.append(color)
+        elif isinstance(color, str):
+            colors = [color] * self.number_of_components()
+        else:
+            colors = color
+
+        combined_plot = Graphics()
+        for i, vertex in self._polygon_vertices().items():
+            points = list(vertex)
+            plot = Polyhedron(vertices=points).plot(color=colors[i])
+            combined_plot += plot
+        return combined_plot
+
+    def _repr_(self):
+        """
+        Return a string representation of ``self``.
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x,y,z> = PolynomialRing(T)
+            sage: (x^4+z^2).tropical_variety()
+            Tropical surface of 0*x^4 + 0*z^2
+        """
+        return f"Tropical surface of {self._poly}"
+
+    def _latex_(self):
+        r"""
+        Return a latex representation of ``self``.
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ)
+            sage: R.<x,y,z> = PolynomialRing(T)
+            sage: tv = (x^2 + R(-1)*x*y*z + R(1)).tropical_variety()
+            sage: latex(tv)
+            Tropical surface of \left(-1\right) x y z + 0 x^{2} + 1
+        """
+        return f"Tropical surface of {self._poly._latex_()}"
```

Comment:
This is not latex output. Something like `TV({self._poly._latex_()})` would be much better. Choose something that seems like it might be common or at least suggestive of what it is. This also can be done in the base class only, no need for special notation for surfaces/curves.